scrapy/core/engine.py

```diff
@@ -205,7 +206,19 @@ def _next_request(self) -> None:
                     extra={"spider": self.spider},
                 )
             else:
-                self.crawl(request)
+                if isinstance(request, Request):
+                    self.crawl(request)
+                elif is_item(request):
+                    self.scraper.slot.itemproc_size += 1
+                    dfd = self.scraper.itemproc.process_item(
+                        request, self.crawler.spider
+                    )
+                    dfd.addBoth(
+                        self.scraper._itemproc_finished,
+                        request,
+                        f"{self.crawler.spider.__class__.__name__}.start_requests",
+                        self.crawler.spider,
+                    )
```

Comment:
Looks good to me.

I wonder if maybe we should fallback to self.crawl, i.e. add else with self.crawl, to maintain backward compatibility.

I also wonder if the elif code could be wrapped into a method, if that way we can avoid any code duplication, but otherwise this looks fine to me.

We do need tests and mention this, the ability to yield items, in the reference docs of the start_requests method.

Reply:
There is also process_start_requests in spider middlewares, at least its doc needs to be updated.

Reply:
> We do need tests and to mention this (the ability to yield items) in the reference docs of the start_requests method.

> There is also process_start_requests in spider middlewares, at least its doc needs to be updated.

Added as result of https://github.com/scrapy/scrapy/pull/6417/commits/8f24f131a64914963c8d95d81f972087cd7f239d